### PR TITLE
Panel: Add new Panel prop `isScrollable`

### PR DESCRIPTION
**Pull request by ogupte**
## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [x] IE11 (Win 7)
  - [x] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [x] Two core team engineer approvals
- [x] One core team UX approval

### CODE REVIEW

**Comment by mute** [comment]
Is this fixing #651 also?

**Comment by ogupte** [comment]
re: https://github.com/appnexus/lucid/pull/653#issuecomment-265506596

Yea in the process of cleaning up the LESS, it happens to fix #651. I didn't intend to fix it but good catch.

**Comment by ogupte** [comment]
re: https://github.com/appnexus/lucid/pull/653#issuecomment-265506596

I don't think it's very valuable to have an example with `isScrollable={false}`. It only exists as a prop for backward compat reasons, but in the next major, all overflowing content should be scrollable in Panel, and this prop should be removed.

File: src/components/Panel/Panel.less [hunk 1]
```diff
@@ -6,23 +6,36 @@
 	background-color: @color-white;
 	border: @border-standardBorder;
 
-	&-has-margin {
+	.lucid-Panel-has-margin& {
```
Comment (jondlm) [inline src/components/Panel/Panel.less]: I think we typically do `&&-has-margin`

File: src/components/Panel/Panel.less [hunk 2]
```diff
@@ -6,23 +6,36 @@
...
+	& > &-Header {
 		padding: @size-standard;
 		border-bottom: @border-lightBorder;
 	}
 
-	&-is-not-gutterless &-content {
-		padding: @size-standard;
-		overflow: auto;
+	& > &-content {
+		.lucid-Panel-is-not-gutterless& {
```
Comment (jondlm) [inline src/components/Panel/Panel.less]: This rule is confusing as well, it equates to this at the root level: 

```less
&&-is-not-gutterless > &-content {
    overflow: auto;
}
```

Comment (ogupte) [inline src/components/Panel/Panel.less]: This is following the LESS docs to select for parent classes (http://lesscss.org/features/#parent-selectors-feature-changing-selector-order).

For example we could have three different selectors with `&-content` in three different places, which resembles  vanilla CSS more closely:

```
& > &-content {
  //default styles
}
&&-is-not-gutterless > &-content {...}
&&-is-scrollable > &-content {...}
```

Or, we can have one section for content with defaults styles with a parent class selector to extend them:

```
& > &-content {
  //default content styles

  .lucid-Panel-is-not-gutterless& {...}
  .lucid-Panel-is-scrollable& {...}
}
```

Both approaches produce the same CSS, but the latter is better for codeorg by making use of available LESS features. I've used this pattern in other components in this repo, and feel we should be making more use of it.

File: src/components/Panel/Panel.less (hunk 1 above)
Comment (ogupte) [inline src/components/Panel/Panel.less]: will fix

File: src/components/Panel/Panel.less (hunk 2 above)
Comment (jondlm) [inline src/components/Panel/Panel.less]: Ahh. I didn't know about that feature. Sounds good!

**Pull request merged**